### PR TITLE
Fix #724

### LINKED ISSUE #274: Add gammapy.catalog.GalacticLatitudeDistribution class
We should add a GalacticLatitudeDistribution class that has methods for:
- [ ] quick plot (see [here](http://astropy.readthedocs.org/en/latest/visualization/histogram.html))
- [ ] density estimation and bootstrap error bands
- [ ] simple model fit (e.g. gauss + const) and error propagation to get an error band

Thoughts and ideas for other features welcome!

(I started experimenting with this in a [notebook](http://nbviewer.ipython.org/gist/cdeil/fcf560bb4d906c251459))

### LINKED ISSUE #724: TestFluxEstimator.test_with_power_law fail in master
@joleroi - I think after your changes yesterday `TestFluxEstimator.test_with_power_law` fails in master. Looks like the energy grouping isn't applied to `backscale_ratio`?

```
E       TypeError: statistic input array sizes do not match
```

https://travis-ci.org/gammapy/gammapy/jobs/162223291#L938

I will mark that test as `xfail` for now.

Putting this under the 0.5 milestone.
Let me know if you don't have time, and I'll try to do it on Monday.

### PR DESCRIPTION
**Pull request by joleroi**
I cannot reproduce any of the fails described in https://github.com/gammapy/gammapy/issues/724 locally (with stable sherpa version)

let's see what travis does

EDIT: Apparently the sherpa conda package  was updated @cdeil Do you know a way to keep track of such things? Like a notification or something.

### CODE REVIEW

**Comment by cdeil** [comment]
You can subscribe to the feed at http://pysherpa.blogspot.de/
Sometimes they announce something there, e.g. for the Sherpa 4.8.2 release they did.

Otherwise I'm not aware of RSS feeds for Sherpa updates.

http://www.astropython.org/packages/sherpa35/ doesn't provide this I think.
Neither does https://anaconda.org/sherpa/sherpa/files

Do you know how to fix the issue.
In case it matters: IMO we should only support Sherpa 4.8.2 and require Gammapy users to update.

**Comment by joleroi** [comment]
Thanks!

> In case it matters: IMO we should only support Sherpa 4.8.2 and require Gammapy users to update.

agreed, unfortunately the WSTAT issue has not been fixed in 4.8.2.. Let's discuss offline if it makes sense to introduce a fitter based on MINUIT at least for simple fits like flux points.

**Comment by joleroi** [comment]
https://travis-ci.org/gammapy/gammapy/jobs/163657473#L1145

conda installs an old sherpa version on travis (should be 4.8.2)
TODO: Fix

**Comment by cdeil** [comment]
Looks like Sherpa `4.8.0-py27_2` is fetched from the Sherpa channel:
https://travis-ci.org/gammapy/gammapy/jobs/163657473#L574

This is one of the files listed here:
https://anaconda.org/sherpa/sherpa/files

What we want is that Sherpa 4.8.2 is installed instead, which is also available there:
`linux-64/sherpa-4.8.2-py27_2.tar.bz2`

I don't understand why we're not getting this new Sherpa from conda.
Our [.travis.yml][https://github.com/gammapy/gammapy/blob/master/.travis.yml] only mentions the Sherpa conda channel and Sherpa, no version number.
Wo we should get the latest stable version, 4.8.2, no?

@olaurino @bsipocz - Any idea what the issue is?

**Comment by olaurino** [comment]
I never used the `CONDA_CHANNELS` environment variable as I remember having issues with it. Should I be able to install sherpa by `export CONDA_CHANNELS=sherpa; conda install sherpa`? That doesn't seem to work locally, but `conda install -c sherpa sherpa` does, and it points to the correct version (conda version 4.1.12). I don't know if this issue is related: 
https://github.com/conda/conda/pull/3390

As for notifications that a new version of sherpa is available, there is an atom feed in GitHub:
https://github.com/sherpa/sherpa/releases.atom

Maybe we should advertise it, @DougBurke and @anetasie?

**Comment by cdeil** [comment]
@olaurino - Thanks.
I'm not sure how to do the normal `conda install -c sherpa sherpa` with our `.travis.yml` file.
@bsipocz - Maybe you have an idea?

> Maybe we should advertise it, @DougBurke and @anetasie?

+1 to advertise Sherpa releases, especially a major achievement like Python 3 support.
`astropy` and / or `astropy-dev` mailing list and http://astropython.org/ come to mind.

**Comment by olaurino** [comment]
To clarify, I meant advertising the GitHub atom feed, which seems more reliable for people like you who use sherpa as a dependency, in our docs. I am sure @anetasie will appreciate the suggestion of additional advertising channels, though.

**Comment by olaurino** [comment]
> agreed, unfortunately the WSTAT issue has not been fixed in 4.8.2.. Let's discuss offline if it makes sense to introduce a fitter based on MINUIT at least for simple fits like flux points.

@cdeil @joleroi Is there anything we can do to help you working with a version of Sherpa that has the `wstat` issue fixed?

In all likelihood the fix will be in the next standalone release, but that will be in several weeks.

If that's blocking your work you could build a sherpa binary from @joleroi's PR. If we can support you in any way to remove this roadblock please let me know.

**Comment by DougBurke** [comment]
@olaurino - I don't think you mean that the next standalone release of
sherpa is in a few weeks. Unless I'm missing something

Doug

On Mon, Oct 3, 2016, 13:57 Omar Laurino <notifications@github.com> wrote:

agreed, unfortunately the WSTAT issue has not been fixed in 4.8.2.. Let's
discuss offline if it makes sense to introduce a fitter based on MINUIT at
least for simple fits like flux points.

@cdeil <https://github.com/cdeil> @joleroi <https://github.com/joleroi> Is
there anything we can do to help you working with a version of Sherpa that
has the wstat issue fixed?

In all likelihood the fix will be in the next standalone release, but that
will be in several weeks.

If that's blocking your work you could build a sherpa binary from @joleroi
<https://github.com/joleroi>'s PR. If we can support you in any way to
remove this roadblock please let me know.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/gammapy/gammapy/pull/729#issuecomment-251177383>, or mute
the thread

**Comment by cdeil** [comment]
@olaurino - Well yes, from our side the sooner the WSTAT fix can go in Sherpa master and then a released version, the better. But I realise it takes time to figure out what the proper fix for a given issue is, add tests, so I fully understand that changes in Sherpa are done more carefully than for Gammapy where everything is new and we don't support many users yet. For now, I'm using Sherpa master with @joleroi's commit cherry-picked for my science analysis. :-)

If you already have a rough plan of when Sherpa releases will happen, it would be nice if you could mention the schedule somewhere, e.g. at  https://github.com/sherpa/sherpa#history .

Especially for Gammapy 1.0 (date not set, maybe December 2016) we'll have to see whether there's a Sherpa release available where WSTAT works for us, or whether for now we'll just change our implementation and call [gammapy.stats.wstat](https://gammapy.readthedocs.io/en/latest/api/gammapy.stats.wstat.html) to compute the fit statistic and "only" use Sherpa for the `fit`, `covar` and `conf` functionality, i.e. as fitting engine. The same question is there for the joint fit  with different per-dataset stats mentioned in a different issue in the Sherpa tracker.

**Comment by joleroi** [comment]
> Well yes, from our side the sooner the WSTAT fix can go in Sherpa master and then a released version, the better

I am using https://github.com/sherpa/sherpa/pull/269 locally for my analyses.

**Comment by DougBurke** [comment]
@cdeil we merged https://github.com/sherpa/sherpa/pull/195 which includes a test which is supposed to provide an example of using different stats per dataset, but having looked at it (with regard to the wstat bug) I can understand if it's hard to follow. Do you have any documentation on how you would like it to work?

I think it's fair to say that the approach used in the PR above is a hack, since there's an implicit assumption that a single statistic is being used in many parts of the sherpa.fit.Fit machinery (and more so in the ui layer). I'm wondering whether we can improve upon the "hack" approach - but keep it as a hack - since I doubt we have the effort to doing anything more extensive on any useful timescale for GammaPy.

**Comment by bsipocz** [comment]
> I never used the CONDA_CHANNELS environment variable as I remember having issues with it. Should I be able to install sherpa by export CONDA_CHANNELS=sherpa; conda install sherpa? That doesn't seem to work locally, but conda install -c sherpa sherpa does, and it points to the correct version (conda version 4.1.12). I don't know if this issue is related: 

Now I see that the choice of variable names in ``ci-helpers`` might not be the best. We use ``CONDA_CHANNELS`` there, too and add all the channels listed there manually using ``conda config --add channels``. I've never tried to use it in any other way.

**Comment by bsipocz** [comment]
@cdeil - Actually I don't know the exact reason of why the old version was picked up. Usually it's because of some version number conflicts.
So I think the most probable candidate is that the new 4.8.2 build requires ``[u'libgfortran 3.0.0', u'libgcc 4.8.5', u'numpy >1.9', u'pytest', u'python 3.5*', u'setuptools', u'six']``, while there is a version pinning in force for 1.0.x.

However this latter got removed in #737, and indeed the builds for that PR pick up the newer sherpa version.

**Comment by cdeil** [comment]
@bsipocz - Indeed, I see that in #737 the build picked up Sherpa 4.8.2:
https://travis-ci.org/gammapy/gammapy/jobs/166537729#L1151

But now, after re-starting the build for this PR after merging #737, I still see Sherpa 4.8.0 being picked up:
https://travis-ci.org/gammapy/gammapy/jobs/163657473#L1158

@bsipocz - Any idea why this is happening?

@joleroi - Could you please rebase this PR on master and force push it again? I don't see why, but maybe this will fix the issue?

**Comment by bsipocz** [comment]
@cdeil - this needs rebasing to pick up the changes from #737

**Comment by joleroi** [comment]
done

@cdeil @bsipocz @DougBurke @olaurino  thanks a lot for your help!
